sql-plugin/src/main/scala/com/nvidia/spark/ExclusiveModeGpuDiscoveryPlugin.scala

```diff
@@ -35,7 +35,7 @@ import org.apache.spark.resource.{ResourceInformation, ResourceRequest}
  *  This plugin can be activated in spark with the configuration:
  *  `--conf spark.resources.discoveryPlugin=com.nvidia.spark.ExclusiveModeGpuDiscoveryPlugin`
  */
-class ExclusiveModeGpuDiscoveryPlugin extends ResourceDiscoveryPlugin with Proxy {
+protected class ExclusiveModeGpuDiscoveryPlugin extends ResourceDiscoveryPlugin with Proxy {
```

Comment:
This is the public class that the user needs to use [per the documentation](https://github.com/NVIDIA/spark-rapids/blob/branch-22.02/docs/get-started/getting-started-on-prem.md#yarn-without-isolation).  Any class directly under `com.nvidia.spark`, as opposed to under `com.nvidia.sparki.rapids`, should be public.

Reply:
Why is this change a part of this PR?

Reply:
Thanks @jlowe and @razajafri. I have kept it as public now. I misunderstood the bug and made this change.